wingtips-lightstep/src/main/java/com/nike/wingtips/lightstep/WingtipsToLightStepLifecycleListener.java

```diff
@@ -0,0 +1,152 @@
+
+    // we borrowed the logging and exception log rate limiting from the Zipkin plugin.
+    private final Logger lightStepToWingtipsLogger = LoggerFactory.getLogger("LIGHTSTEP_SPAN_CONVERSION_OR_HANDLING_ERROR");
+
+    private final AtomicLong spanHandlingErrorCounter = new AtomicLong(0);
+    private long lastSpanHandlingErrorLogTimeEpochMillis = 0;
+    private static final long MIN_SPAN_HANDLING_ERROR_LOG_INTERVAL_MILLIS = TimeUnit.SECONDS.toMillis(60);
+
+    protected String serviceName;
+    protected String accessToken;
+    protected String satelliteUrl;
+    protected int satellitePort;
+
+    protected Tracer tracer = null;
+
+    // Basic constructor which requires values to configure tracer and point span traffic from the transport library
+    // to the LightStep Satellites.
+    public WingtipsToLightStepLifecycleListener(String serviceName, String accessToken, String satelliteUrl, int satellitePort) {
+        this.serviceName = serviceName;
+        this.accessToken = accessToken;
+        this.satelliteUrl = satelliteUrl;
+        this.satellitePort = satellitePort;
+
+        try {
+            this.tracer = new com.lightstep.tracer.jre.JRETracer(
+                    new com.lightstep.tracer.shared.Options.OptionsBuilder()
+                            .withAccessToken(this.accessToken)
+                            .withComponentName(this.serviceName)
+                            .withCollectorHost(this.satelliteUrl)
+                            .withCollectorPort(this.satellitePort)
+                            .withVerbosity(4)
+                            .build()
+            );
+        } catch (Throwable ex) {
+            lightStepToWingtipsLogger.warn(
+                    "There has been an issue with initializing the LightStep tracer: ", ex.toString());
+        }
+    }
+
+    @Override
+    public void spanSampled(Span wingtipsSpan) {
+        // Do nothing
+    }
+
+    @Override
+    public void spanStarted(Span wingtipsSpan) {
+        // Do nothing
+    }
+
+    @Override
+    public void spanCompleted(Span wingtipsSpan) {
+        String operationName = wingtipsSpan.getSpanName();
+        long startTimeMicros = wingtipsSpan.getSpanStartTimeEpochMicros();
+
+        // Given we should only be in this method on span completion, we are not going to wrap this conversion in a
+        // try/catch. duration should be set on the Wingtips span.
+        long durationMicros = TimeUnit.NANOSECONDS.toMicros(wingtipsSpan.getDurationNanos());
+        long stopTimeMicros = startTimeMicros + durationMicros;
+
+        // parentId will get changed to reflect the Wingtips parent id. If there is no id then a value of 0 will get
+        // converted into null on the LightStep Satellite. LightStep will require our Ids to be in long format.
+        long lsParentId = 0;
+        long lsSpanId = TraceAndSpanIdGenerator.unsignedLowerHexStringToLong(wingtipsSpan.getSpanId());
```

Comment:
The Trace ID, Span ID, and Parent ID might all need sanitization. We get requests with IDs that don't follow the expected format all the time, and it's traffic that we can't just ignore.

See how this was solved in the zipkin module [here](https://github.com/Nike-Inc/wingtips/blob/751b687320623b1b258cdcd9594df49180375385/wingtips-zipkin2/src/main/java/com/nike/wingtips/zipkin2/util/WingtipsToZipkinSpanConverterDefaultImpl.java#L61-L63), [here](https://github.com/Nike-Inc/wingtips/blob/751b687320623b1b258cdcd9594df49180375385/wingtips-zipkin2/src/main/java/com/nike/wingtips/zipkin2/util/WingtipsToZipkinSpanConverterDefaultImpl.java#L81-L89) (adding correlation tags when ID sanitization was required), and [here](https://github.com/Nike-Inc/wingtips/blob/751b687320623b1b258cdcd9594df49180375385/wingtips-zipkin2/src/main/java/com/nike/wingtips/zipkin2/util/WingtipsToZipkinSpanConverterDefaultImpl.java#L123-L172) (the sanitization code).

If you don't do this, then a large chunk of our traffic will fail here, as the IDs we receive won't be convertible to longs.

I will probably move some of that code to a utils class so it can be reused, but for now you can copy/paste whatever's necessary.